pkg/concurrencylimiter/limiter.go

```diff
@@ -0,0 +1,61 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package concurrencylimiter
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+)
+
+// Limiter limits concurrency of running tasks.
+type Limiter struct {
```

Comment:
Seems like this could just be https://godoc.org/golang.org/x/sync/semaphore

Reply:
Cool. I didn't know there is such package, this is exactly what I want at the beginning. Didn't find it it golang `sync`, thus implemented one with channel myself.

Will use this instead.

Reply:
Done